### PR TITLE
Timeout: Fix #4281 Add timeout param to junos_config module

### LINKED ISSUE #4281: junos modules RPC timeout option
##### ISSUE TYPE
- Feature Idea
##### COMPONENT NAME

junos_config
##### ANSIBLE VERSION

```
ansible 2.1.0.0
```
##### OS / ENVIRONMENT

```
Linux box 3.16.0-67-generic #87~14.04.1-Ubuntu SMP Fri Mar 11 00:26:02 UTC 2016 x86_64 x86_64 x86_64 GNU/Linux
```

```
Juniper EX2200 switches (12.3R8.7), some are virtual chassis
```
##### SUMMARY

Default 30sec RPC timeout breaks commits on EX virtual chassis.
Need an explicit timeout option like in [junos_install_config](http://junos-ansible-modules.readthedocs.io/en/1.3.0/junos_install_config.html)
##### STEPS TO REPRODUCE

```
playbook.yml 

---
- name: set config
  hosts: juniper_test
  connection: local
  gather_facts: no

  vars_prompt:
   - name: USERNAME
     prompt: Username
     private: no
   - name: DEVICE_PASSWORD
     prompt: Password
     private: yes

  tasks:
  - name: set config
    junos_config:
      host: "{{ inventory_hostname }}"
      username: "{{ USERNAME }}"
      port: 22
      password: "{{ DEVICE_PASSWORD }}"
      lines:
        - set system login class monitoring permissions secret
        - set system login class monitoring permissions view
        - set system login class monitoring permissions view-configuration
```
##### ACTUAL RESULTS

[playbook run](https://gist.github.com/dmitrijsf/3abe1642f297b81750448c2e4d7b93b1)

### PR DESCRIPTION
**Pull request by ganeshrn**
##### ISSUE TYPE

 - Feature Pull Request

##### COMPONENT NAME
junos_config

##### ANSIBLE VERSION

```
ansible 2.3.0 (devel 461286f914)
```

##### SUMMARY

```
Add NETCONF RPC timeout beyond the default value of
30 seconds. Set this value to accommodate configuration
changes (commits) that might take longer than the default
timeout interval.
```

### CODE REVIEW

**Comment by gregdek** [comment]
Thanks @ganeshnalawade for this PR. This PR requires revisions, either because it fails to build or by reviewer request. Please make the suggested revisions. When you are done, please comment with text 'ready_for_review' and we will put this PR back into review.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by ganeshrn** [comment]
@gregdek ready_for_review

**Comment by gregdek** [comment]
Thanks @ganeshnalawade. To the current maintainers, @privateip, @gundalow, @qalthos please review according to guidelines (http://docs.ansible.com/ansible/developing_modules.html#module-checklist) and comment with text 'shipit', 'needs_revision' or 'close_me' as appropriate.

[This message brought to you by your friendly Ansibull-bot.]

File: network/junos/junos_config.py [hunk 1]
```diff
@@ -113,6 +113,15 @@
     default: no
     choices: ['yes', 'no']
     version_added: "2.2"
+  timeout:
```
Comment (gundalow) [inline network/junos/junos_config.py]: netconf_timeout

**Review by gundalow — CHANGES_REQUESTED** [review]
needs_revision

**Comment by ganeshrn** [comment]
As per @Qalthos will check if netconf timeout can be implemented in module_utils so that it can be added to all other junos modules

**Comment by ganeshrn** [comment]
@gundalow @Qalthos ready_for_review
Related PR https://github.com/ansible/ansible/pull/18616

**Comment by ganeshrn** [comment]
Closing this PR, as https://github.com/ansible/ansible/pull/18634 fixes the issue

**Pull request closed without merging**